### DISCUSSION TITLE
Encryption

### DISCUSSION THREAD

**Discussion by tcurdt** [k8up-io/k8up#972] [Q&A] [answered]
Is there support for server side encryption?

My ideal setup would
- create a new key for every backup
- use asymmetric encryption to encrypt that key and store it with the backup
- upload to S3
- require the private key to decrypt the backup key on restore

I know restic does support encryption. Is there at least support for that?
I somehow cannot find anything about this in the docs.

**Comment by tobru** [comment, marked as answer]
K8up is using Restic encryption by default. When specifiying a [Backup](https://docs.k8up.io/k8up/2.10/references/object-specifications.html#_backup) object, the `spec.backend.repoPasswordSecretRef` references a secret which contains the password to encrypt the data before sending to the target.
